mathics/builtin/inout.py

```diff
@@ -212,7 +213,8 @@ def parenthesize(precedence, element, element_boxes, when_equal):
     elif element.has_form("PrecedenceForm", 2):
         element_prec = element.elements[1].get_int_value()
     # For negative values, ensure that the element_precedence is at least the precedence. (Fixes #332)
-    elif isinstance(element, Integer) and element.value < 0:
+    elif isinstance(element, (Integer, Real)) and element.value < 0:
```

Comment:
And Complex literal?

Reply:
``Complex`` and ``Rational`` seems to be handled properly without any change:
```
In[1]:= (2+3 I)^q
Out[1]= (2 + 3 I) ^ q

In[2]:= (2/3)^q
Out[2]= (2 / 3) ^ q
```